gradoop-flink/src/main/java/org/gradoop/flink/datagen/foodbroker/FoodBroker.java

```diff
@@ -0,0 +1,272 @@
+   * Foodbroker configuration.
+   */
+  private final FoodBrokerConfig foodBrokerConfig;
+
+  /**
+   * Set which contains all customer vertices.
+   */
+  private DataSet<Vertex> customers;
+  /**
+   * Set which contains all vendor vertices.
+   */
+  private DataSet<Vertex> vendors;
+  /**
+   * Set which contains all logistic vertices.
+   */
+  private DataSet<Vertex> logistics;
+  /**
+   * Set which contains all employee vertices.
+   */
+  private DataSet<Vertex> employees;
+  /**
+   * Set which contains all product vertices.
+   */
+  private DataSet<Vertex> products;
+  /**
+   * Set which contains one map from the gradoop id to the quality of a customer vertex.
+   */
+  private DataSet<Map<GradoopId, Float>> customerDataMap;
+  /**
+   * Set which contains one map from the gradoop id to the quality of a vendor vertex.
+   */
+  private DataSet<Map<GradoopId, Float>> vendorDataMap;
+  /**
+   * Set which contains one map from the gradoop id to the quality of a logistic vertex.
+   */
+  private DataSet<Map<GradoopId, Float>> logisticsDataMap;
+  /**
+   * Set which contains one map from the gradoop id to the quality of an employee vertex.
+   */
+  private DataSet<Map<GradoopId, Float>> employeesDataMap;
+  /**
+   * Set which contains one map from the gradoop id to the quality of a product vertex.
+   */
+  private DataSet<Map<GradoopId, Float>> productsQualityDataMap;
+  /**
+   * Set which contains one map from the gradoop id to the price of a product vertex.
+   */
+  private DataSet<Map<GradoopId, BigDecimal>> productsPriceDataMap;
+
+
+  /**
+   * Valued constructor.
+   *
+   * @param env execution environment
+   * @param gradoopFlinkConfig Gradoop Flink configuration
+   * @param foodBrokerConfig Foodbroker configuration
+   */
+  public FoodBroker(ExecutionEnvironment env, GradoopFlinkConfig gradoopFlinkConfig,
+    FoodBrokerConfig foodBrokerConfig) {
+
+    this.env = env;
+    this.gradoopFlinkConfig = gradoopFlinkConfig;
+    this.foodBrokerConfig = foodBrokerConfig;
+  }
+
+  @Override
+  public GraphCollection execute() {
+    // used for type hinting when loading graph head data
+    TypeInformation<GraphHead> graphHeadTypeInfo = TypeExtractor
+      .createTypeInfo(gradoopFlinkConfig.getGraphHeadFactory().getType());
+    // used for type hinting when loading vertex data
+    TypeInformation<Vertex> vertexTypeInfo = TypeExtractor
+      .createTypeInfo(gradoopFlinkConfig.getVertexFactory().getType());
+    // used for type hinting when loading edge data
+    TypeInformation<Edge> edgeTypeInfo = TypeExtractor
+      .createTypeInfo(gradoopFlinkConfig.getEdgeFactory().getType());
+
+    // Phase 1: Create MasterData
+    initMasterData();
+
+    // Phase 2.1: Run Brokerage
+    DataSet<Long> caseSeeds = env.generateSequence(1, foodBrokerConfig
+      .getCaseCount());
+
+    DataSet<GraphTransaction> brokerage = caseSeeds
+      .mapPartition(new Brokerage(gradoopFlinkConfig.getGraphHeadFactory(), gradoopFlinkConfig
```

Comment:
Why still mapParition? To which degree depend cases on each other? Should be independent.